crates/zksync/core/src/vm/inspect.rs

```diff
@@ -222,6 +229,9 @@ where
         "gas usage",
     );
 
+    let account_accesses = era_db.get_account_accesses();
+
+    // TODO(zk): adapt this to use account_accesses as well
```

Comment:
do we really need to propagate account accesses as part of the observable state? I mean `modified_storage` should cover everything, right? Anything in account accesses should have a corresponding entry in modified storage, no?

Reply:
Modified storage does not account for reads. In fact account accesses is a super set of modified storage and contains a record for all `CALL`s, `CREATE`s _and_ storage accesses (read and write).

Reply:
Yes, but this section is about state persistence, which reads shouldn't affect, right?